lib/src/results.dart

```diff
@@ -82,15 +87,17 @@ class RealmResults<T extends RealmObjectBase> extends collection.IterableBase<T>
     final controller = ResultsNotificationsController<T>(this);
     return controller.createStream();
   }
+}
 
-  /// Creates a frozen snapshot of this query.
-  RealmResults<T> freeze() {
-    if (isFrozen) {
-      return this;
-    }
-
-    final frozenRealm = realm.freeze();
-    return frozenRealm.resolveResults(this);
+// The query operations on results only work for results of objects (core restriction),
+// so we add it as an extension methods to allow the compiler to prevent misuse.
+extension RealmResultsOfObject<T extends RealmObjectBase> on RealmResults<T> {
```

Comment:
```suggestion
extension RealmResultsOfObject<T extends RealmObject> on RealmResults<T> {
```
It shouldn't be allowed for embedded objects, I think.
Can you make a query on a RealmResults of embedded objects?

Reply:
Yes you can, see `embedded_test.dart:724` as an example

Reply:
If query is working only for RelmObjects then we have to change also tha same extension on RealmList<T>.

Reply:
Ah, OK.